dart-impl/mpi/src/dart_communication.c

```diff
@@ -30,6 +30,11 @@
 #include <math.h>
 #include <alloca.h>
 
+/* the number of consecutive memcpy for local put/get before calling into MPI */
+#define NUM_CONSECUTIVE_MEMCPY 16
+
+/* number of performed local memcpy between calling into MPI */
+static _Thread_local int num_local_memcpy = 0;
```

Comment:
this needs C11, is this already ensured by CMake? I also think its better to use `<threads.h>` and `thread_local` here.

Reply:
and no `= 0` needed

Reply:
You're right, we don't officially require a C11 compiler. I am not going to open the discussion on whether  we should abandon support for >20 year old compilers though...